sdk/core/azure-core-amqp/inc/azure/core/amqp/network/transport.hpp

```diff
@@ -16,60 +17,119 @@ namespace Azure { namespace Core { namespace Amqp { namespace Network { namespac
   };
 
   class Transport;
 
+  /** Transport operational events.
+   *
+   * Normally a client will not have to register for these events, but they are available for
+   * clients if necessary.
+   */
   struct TransportEvents
   {
-    virtual void OnOpenComplete(TransportOpenResult openResult) = 0;
+    /** @brief Called when bytes are received on the transport.
+     *
+     * @param transport The transport on which bytes are received.
+     * @param buffer The buffer containing the bytes received.
+     * @param size The size of the buffer.
+     *
+     */
     virtual void OnBytesReceived(
         Transport const& transport,
         const unsigned char* buffer,
         size_t size)
         = 0;
+
+    /** @brief Called when an error has occurred on the transport */
     virtual void OnIoError() = 0;
   };
 
-  class Transport {
+  class Transport final {
 
   public:
-    using TransportCloseCompleteFn = std::function<void()>;
-    using TransportSendCompleteFn = std::function<void(TransportSendResult)>;
-    Transport(std::shared_ptr<_detail::TransportImpl> impl) : m_impl{impl} {}
-    virtual ~Transport();
-    virtual bool Open();
-    virtual bool Close(TransportCloseCompleteFn);
-    virtual bool Send(uint8_t*, size_t, TransportSendCompleteFn) const;
+    using TransportSendCompleteFn = std::function<void(TransportSendStatus)>;
+
+    /** @brief Called when a transport object is destroyed. */
+    ~Transport();
+
+    /** @brief Open the transport.
+     *
+     * @param context The context for cancellation.
+     * @return TransportOpenStatus
+     */
+    TransportOpenStatus Open(Azure::Core::Context const& context = {});
+
+    /** @brief Close the transport.
+     *
+     * @param context - The context for cancellation.
+     *
+     */
+    void Close(Azure::Core::Context const& context = {});
+
+    /** @brief Send bytes on the transport.
+     *
+     * @param buffer The buffer containing the bytes to send.
+     * @param size The size of the buffer.
+     * @param callback The callback to be called when the send operation is completed.
+     * @return bool
+     */
+    bool Send(uint8_t* buffer, size_t size, TransportSendCompleteFn callback) const;
+
+    /** @brief Poll the transport for events.
+     */
     void Poll() const;
-    virtual std::shared_ptr<_detail::TransportImpl> GetImpl() const { return m_impl; }
+
+    /** @brief Set the event handler for the transport.
+     *
+     * @param events The event handler to be set.
+     *
+     * @remarks Normally a client will not have to register for these events via this method, but
+     * there may be circumstances where it is necessary (if, for instance the transport was created
+     * internally as a result of a socket listener).
+     */
     void SetEventHandler(TransportEvents* events);
 
+    Transport(std::shared_ptr<_detail::TransportImpl> impl) : m_impl{impl} {}
+    std::shared_ptr<_detail::TransportImpl> GetImpl() const { return m_impl; }
+
   protected:
```

Comment:
given that the class is `final`, `protected` is confusing.

Reply:
It's an artifact of when the Transport was a base class.